src/Http/Routing/src/ValidationEndpointFilterFactory.cs

```diff
@@ -28,6 +29,10 @@ public static EndpointFilterDelegate Create(EndpointFilterFactoryContext context
             return next;
         }
 
+        var jsonOptions = context.ApplicationServices.GetService<IOptions<JsonOptions>>();
+        var serializerOptions = jsonOptions?.Value?.SerializerOptions;
+        options.SerializerOptions = serializerOptions;
```

Comment:
Are we at all concerned about anyone else grabbing `IOptions<ValidationOptions>` from DI and setting a different `SerializerOptions`? Or someone setting `SerializerOptions` at app startup but then we overwrite it here? Or someone using `IOptions<ValidationOptions>` elsewhere and us setting the `SerializerOptions` property?